registry/remote/auth/example_test.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// Package remote_test includes all the testable examples for remote repository type
+
+package auth_test
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+const (
+	username    = "test_user"
+	password    = "test_password"
+	accessToken = "test/access/token"
+)
+
+var host string
+var simpleDir string
+var basicAuthDir string
+var accessTokenDir string
+var clientConfigDir string
+var scopes = []string{
+	"repository:dst:pull,push",
+	"repository:src:pull",
+}
+
+func TestMain(m *testing.M) {
+	// create an authorization server
+	as := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
The authorization server should be used for refresh token example

Reply:
The tests in client_test.go covers possible cases comprehensively. I don't want these example tests to be a duplicate of those, so I only show the simple and common cases. The mock server also handles the requests in a simplified manner.